sdk/core/System.ClientModel/src/Convenience/ClientConnection.cs

```diff
@@ -59,6 +62,27 @@ internal ClientConnection(string id, string locator, CredentialKind credentialKi
         Id = id;
         Locator = locator;
         CredentialKind = credentialKind;
+        Metadata = new ReadOnlyDictionary<string, string>(new Dictionary<string, string>());
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ClientConnection"/> struct with the specified subclient ID.
+    /// It is only for the JSON serializer.
+    /// </summary>
+    /// <param name="name">The name of a connection.</param>
+    /// <param name="type">The connection type.</param>
+    /// <param name="metadata">The connection metadata.</param>
+    /// <param name="id">The identifier for the connection.</param>
+    /// <param name="locator">The endpoint or resource identifier.</param>
+    /// <param name="credentialKind">The kind of connection used by the client</param>
+    internal ClientConnection(string name, string type, IReadOnlyDictionary<string, string> metadata, string id, string locator, CredentialKind credentialKind)
```

Comment:
What's the difference between "name" and "id"?

Reply:
That appears to be a resource id.. If it is, should we name it `resourceId` and update the description accordingly?

Reply:
Removed.